crates/store/src/state.rs

```diff
@@ -1131,16 +1137,32 @@ async fn load_account_tree(
     db: &mut Db,
     block_number: BlockNumber,
 ) -> Result<AccountTreeWithHistory<InMemoryAccountTree>, StateInitializationError> {
-    let account_data = db.select_all_account_commitments().await?.into_iter().collect::<Vec<_>>();
+    let account_data = db.select_all_account_commitments().await?;
 
-    // Convert account_data to use account_id_to_smt_key
     let smt_entries = account_data
         .into_iter()
         .map(|(id, commitment)| (account_id_to_smt_key(id), commitment));
 
-    let smt = LargeSmt::with_entries(MemoryStorage::default(), smt_entries)
-        .expect("Failed to create LargeSmt from database account data");
+    let smt = LargeSmt::with_entries(MemoryStorage::default(), smt_entries).map_err(|e| {
+        tracing::error!(
+            target: COMPONENT,
+            error = ?e,
+            "Failed to create LargeSmt from database account data"
+        );
```

Comment:
Nit: I would put this in a `.inspect_err`

Reply:
Though come to think, won't this get reported higher up in the call stack? Maybe a debug remnant from dev?